### DISCUSSION TITLE
State of XSS vulnerability?

### DISCUSSION THREAD

**Discussion by sfirke** [apache/superset#21839] [General]
I'm interested in enabling the feature `ENABLE_JAVASCRIPT_CONTROLS` to use custom tooltips like [this](https://stackoverflow.com/a/62181562/4470365).  But there's a comment above that flag's line in config.py, "This exposes an XSS security vulnerability."

Is that still the case?  I see that original warning was [written in Feb 2018](https://github.com/apache/superset/pull/4400/files) and Superset's version of the xss package was [recently updated ](https://github.com/apache/superset/pull/18922) to address security concerns.  I'd like to know if anything has changed since the 2018 warning - it's an area I know little about.

**Comment by eschutho** [comment]
cc @michael-s-molina

**Comment by rusackas** [comment]
This is still the case. The feature allows a user to input arbitrary javascript, which has numerous security implications. There's a bit of JS sandboxing going on, but IIRC the documentation for the sandboxing packages/methodology also say it can't _quite_ be trusted.

**Reply by michael-s-molina** [reply]
👍🏼
